pkg/proxy/service.go

```diff
@@ -171,23 +167,15 @@ func (sct *ServiceChangeTracker) newBaseServiceInfo(port *v1.ServicePort, servic
 		klog.V(4).Infof("service change tracker(%v) ignored the following load balancer source ranges(%s) for service %v/%v as they don't match IPFamily", sct.ipFamily, strings.Join(cidrs, ","), service.Namespace, service.Name)
 	}
 
-	// Obtain Load Balancer Ingress IPs
-	var ips []string
-	for _, ing := range service.Status.LoadBalancer.Ingress {
-		ips = append(ips, ing.IP)
-	}
-
-	if len(ips) > 0 {
-		ipFamilyMap = utilproxy.MapIPsByIPFamily(ips)
+	correctIngresses, incorrectIngresses := utilproxy.FilterIncorrectLoadBalancerIngress(service.Status.LoadBalancer.Ingress, sct.ipFamily)
```

Comment:
We switched from a "filter incorrect" model to a "map by family" model because it is easier to think about and more copyable as a pattern.  Can we convert this to that model, please?